tests/cpp/test_indexing_advanced.cpp

```diff
@@ -753,4 +816,326 @@ TEST_F(NVFuserTest, FusionIndexing18_CUDA) {
   testValidate(fe.kernel(), cg_outputs, inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST_P(AdvancedIndexingTest, 19) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeConcreteTensor({5, 7, 11, 13});
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+
+  auto tv2 = makeConcreteTensor({5, 11});
+  fusion.addInput(tv2);
+
+  auto tv3 = broadcast(tv2, {false, true, false, true});
+  auto tv4 = add(tv3, tv1);
+  fusion.addOutput(tv4);
+
+  // // tv4[5, 7, 11, 13] = tv3[5, b1, 11, b3] + tv1[5, 7, 11, 13]
+  tv4->merge(0, 3);
+  // tv4[5*13, 7, 11]
+  tv4->split(0, 3);
+  // tv4[5*13//3, 3, 7, 11]
+  tv4->merge(2, 3)->split(2, 2);
+  // tv4[5*13//3, 3, 7*11//2, 2]
+  tv4->merge(0, 2);
+  // tv4[(5*13//3)*(7*11//2), 3, 2]
+
+  TransformPropagatorWithCheck propagator(tv4);
+  MaxLogicalDomainInfoSpanningTree(tv4).traverse(&propagator);
+  inlineAllAt(tv4, 1, false);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({5, 7, 11, 13}, options);
+  at::Tensor t1 = at::randn({5, 11}, options);
+  std::vector<c10::IValue> inputs = {t0, t1};
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, inputs);
+  auto outputs = fe.runFusion(inputs);
+
+  testValidate(&fusion, outputs, inputs, __LINE__, __FILE__);
+}
+
+// Create a case where we're missing a valid concrete id so the compute at map
+// processing will fail. We need to be able to create the concrete ID not just
+// look for one.
+TEST_F(AdvancedIndexingIdModelTest, 20) {
```

Comment:
This example seems similar to this example we have looked at previously: https://github.com/NVIDIA/Fuser/pull/2512/files#diff-1e72aab35f54e7aaec2696293a478093301cc0a000d77fba8975cc8571740b84R731-R736. I am wondering why it fails, i.e. which dimension is missing a concrete ID?

Reply:
This fusion has two paths from `tv1`. One is `{tv2, tv3, tv4, tv5`}, and another is `{tv6, tv7, tv8, tv9}`. The first group has `[7, 11, 1]`, while the second has `[7, 1, 13]`. These two groups are inlined together at line 906, so the inlined loops need `[7, 11, 13]`, however, none of the inlined tensors has the complete shape, and `ComputeAtMap` just fails as it can't find a valid concrete domain (at least, it isn't a silent failure).

IdModel's loop promotion can handle this inlining pattern as it automatically create a valid concrete domain. However, since we haven't completely replaced ComputeAtMap with IdModel, the lowering doesn't work yet.